# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/nsCoreUtils.h
# Commit: 4544c8e0eda1
# Full Hash: 4544c8e0eda1a54158ce635db02c449fa4e4338e
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2025-09-30 17:19 +0000
# Regressor Bug: 1987945
# File Overlap Count: 2
# Overlapping Files: accessible/base/nsCoreUtils.cpp, accessible/tests/browser/relations/browser_anchor_positioning.js
# Description:
#   Bug 1987945 - P1: Implement details relationship for CSS anchored local accessibles. r=Jamie,emilio
# ==============================================================================

diff -r b8522d57b4e6 -r 4544c8e0eda1 accessible/base/nsCoreUtils.h
--- a/accessible/base/nsCoreUtils.h	Tue Sep 30 16:58:04 2025 +0000
+++ b/accessible/base/nsCoreUtils.h	Tue Sep 30 17:19:31 2025 +0000
@@ -353,6 +353,24 @@
            aContent->IsGeneratedContentContainerForAfter() ||
            aContent->IsGeneratedContentContainerForMarker();
   }
+
+  /**
+   * Return the anchor frame for the given CSS positioned frame, or null if:
+   * 1. there is none,
+   * 2. there is more than one anchor,
+   * 3. or, there is one or more anchor used for sizing/margin only.
+   */
+  static nsIFrame* GetAnchorForPositionedFrame(
+      const PresShell* aPresShell, const nsIFrame* aPositionedFrame);
+
+  /**
+   * Return the CSS positioned frame for the given anchor frame, or null if:
+   * 1. there is none,
+   * 2. the anchor has more than one positioned frame,
+   * 3. or, there is one or more positioned frame using this anchor for sizing/margin only.
+   */
+  static nsIFrame* GetPositionedFrameForAnchor(const PresShell* aPresShell,
+                                               const nsIFrame* aAnchorFrame);
 };
 
 #endif